src/reachy2_sdk/parts/arm.py

```diff
@@ -385,28 +385,24 @@ def _goto_cartesian_interpolation(
         q1, trans1 = decompose_matrix(origin_matrix)
         q2, trans2 = decompose_matrix(target)
 
-        for t in np.linspace(0, 1, nb_steps):
-            # Linear interpolation for translation
-            trans_interpolated = (1 - t) * trans1 + t * trans2
-
-            # SLERP for rotation interpolation
-            q_interpolated = Quaternion.slerp(q1, q2, t)
-            rot_interpolated = q_interpolated.rotation_matrix
+        if not circular_interpolation:
+            self._send_linear_interpolation(trans1, trans2, q1, q2, nb_steps=nb_steps, time_step=time_step)
 
-            # Recompose the interpolated matrix
-            interpolated_matrix = recompose_matrix(rot_interpolated, trans_interpolated)
-
-            request = ArmCartesianGoal(
-                id=self._part_id,
-                goal_pose=Matrix4x4(data=interpolated_matrix.flatten().tolist()),
+        else:
+            self._send_elliptical_interpolation(
+                trans1,
+                trans2,
+                q1,
+                q2,
+                arc_direction=arc_direction,
+                secondary_radius=secondary_radius,
+                nb_steps=nb_steps,
+                time_step=time_step,
             )
-            self._stub.SendArmCartesianGoal(request)
-            time.sleep(time_step)
 
         current_pose = self.forward_kinematics()
         current_precision_distance_xyz = np.linalg.norm(current_pose[:3, 3] - target[:3, 3])
-        if current_precision_distance_xyz > precision_distance_xyz:
-            print("Precision is not good enough, spamming the goal position!")
+        if precision_first and current_precision_distance_xyz > precision_distance_xyz:
```

Comment:
precision first is a bit redundant with precision_distance. 
Is it still required to do this spam?